cmd/sidecars/kafka_new/client/reader.go

```diff
@@ -0,0 +1,145 @@
+package kafkasc
+
+import (
+	"context"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	globalEnv "github.com/inspr/inspr/pkg/environment"
+	"github.com/inspr/inspr/pkg/ierrors"
+	"github.com/inspr/inspr/pkg/sidecars/models"
+	"go.uber.org/zap"
+)
+
+const pollTimeout = 100
+
+// Reader reads/commit messages from the channels defined in the env
+type Reader struct {
+	consumers map[string]models.Consumer
+}
+
+// NewReader return a new Reader
+func NewReader() (*Reader, error) {
+	var reader Reader
+	channelsList := globalEnv.GetChannelBoundaryList(globalEnv.GetInputChannelsData())
+
+	resolvedChList := globalEnv.GetResolvedBoundaryChannelList(globalEnv.GetInputChannelsData())
+	if len(resolvedChList) == 0 {
+		return nil, ierrors.NewError().Message("INSPR_INPUT_CHANNELS not specified").InvalidChannel().Build()
+	}
+
+	reader.consumers = make(map[string]models.Consumer)
+
+	for idx, ch := range channelsList {
```

Comment:
Is it guaranteed that `channelsList` and `resolvedChList` will be in the same order? Also, calling it "index" is better than "idx"

Reply:
its ordered